src/java/org/apache/cassandra/index/sai/disk/v1/InvertedIndexSearcher.java

```diff
@@ -129,7 +129,7 @@ else if (exp.getOp() == Expression.Op.RANGE)
     }
 
     @Override
-    public CloseableIterator<PrimaryKeyWithSortKey> orderBy(Orderer orderer, AbstractBounds<PartitionPosition> keyRange, QueryContext queryContext, int limit) throws IOException
+    public CloseableIterator<PrimaryKeyWithSortKey> orderBy(Orderer orderer, Expression filter, AbstractBounds<PartitionPosition> keyRange, QueryContext queryContext, int limit) throws IOException
     {
         var iter = new RowIdWithTermsIterator(reader.allTerms(orderer.isAscending()));
```

Comment:
We need to pass the filter in to the reader. Thankfully, this one should be fairly straight forward. I might even have a draft of this part locally. Do you mind if I push up a commit to add it if I find that it is in one of my git stashes?

Reply:
@pkolaczk pointed out offline that we can't do range queries on literals, so we don't need this. (Note: we can do them on maps, but we can't order maps yet, so there is no use pushing down the logic yet.)